radix-engine/src/engine/kernel.rs

```diff
@@ -177,9 +176,12 @@ where
                 ))
             })?;
 
-        if matches!(substate_id, SubstateId::ComponentInfo(..)) {
+        if let SubstateId::ComponentInfo(address) = substate_id {
             node_pointer
-                .acquire_lock(substate_id.clone(), false, false, track)
+                .acquire_lock(SubstateId::ComponentState(*address), false, false, track)
```

Comment:
Not sure if ComponentState needs to be locked here?

Reply:
This is because dynamic loading of component state is not supported yet. In order to create a node representation, we need both the state and info, thus requiring both substates to be locked upfront.

Will be fixed in a follow-up PR. :)